js/src/Components/EntityForm.js

```diff
@@ -36,43 +41,121 @@ export const EntityForm = ({entityType, bundle, entity, onFieldChange, widgetCon
+        {props.header}
+        {Object.keys(formFields.form).map(fieldName => {
+          const field = formFields.form[fieldName];
+          const WidgetName = widgetComponents[field.widget_type];
 
-        if (WidgetName === undefined) {
-          console.error('Unable to find widget for type: ' + field.widget_type);
+          if (WidgetName === undefined) {
+            console.error('Unable to find widget for type: ' + field.widget_type);
+            return (
+              <FieldContainer key={`widget-${fieldName}`}>
+                Unable to provide a form for
+                field {field.label}
+              </FieldContainer>
+            )
+          }
           return (
             <FieldContainer key={`widget-${fieldName}`}>
-              Unable to provide a form for
-              field {field.label}
+              <WidgetName
+                fieldId={fieldName}
+                onFieldChange={onFieldChange.bind(undefined, fieldName)}
+                settings={formFields.form[fieldName]}
+                defaultValue={typeof entity[fieldName] !== 'undefined' ? entity[fieldName] : null}
+                fieldName={fieldName}
+              />
             </FieldContainer>
           )
-        }
-        return (
-          <FieldContainer key={`widget-${fieldName}`}>
-            <WidgetName
-              fieldId={fieldName}
-              onFieldChange={onFieldChange.bind(undefined, fieldName)}
-              settings={formFields.form[fieldName]}
-              defaultValue={typeof entity[fieldName] !== 'undefined' ? entity[fieldName] : null}
-              fieldName={fieldName}
-              bundle={bundle}
-            />
-          </FieldContainer>
-        )
-      })}
+        })}
+        {props.footer}
+      </TabPanel>
+
+
+      {Object.keys(formFields.behavior_plugins).length >= 1 &&
+      <TabPanel value={tabValue} index={1} style={{width: tabPanelWidth}}>
+        <Behaviors
+          behaviors={formFields.behavior_plugins}
+          onBehaviorChange={onBehaviorChange}
+          entityType={entityType}
+          bundle={bundle}
+          entity={entity}
+          itemId={itemId}
+        />
+      </TabPanel>
+      }
     </div>
   )
 
 }
 
+const useStyles = makeStyles((theme) => (
+  {
+    root: {
+      flexGrow: 1,
+      backgroundColor: theme.palette.background.paper,
+      display: 'flex',
+    },
+    tabs: {
+      borderRight: `1px solid ${theme.palette.divider}`,
+      background: '#e6e5e1'
+    },
+    tab: {
+      borderBottom: '1px solid #b3b2ad',
+      background: '#f2f2f0'
+    }
+  }
+));
+
+const TabPanel = (props) => {
+  const {children, value, index, ...other} = props;
+
+  return (
+    <div
+      role="tabpanel"
+      hidden={value !== index}
+      id={`vertical-tabpanel-${index}`}
+      aria-labelledby={`vertical-tab-${index}`}
+      {...other}
+    >
+      {value === index && (
+        <Box p={3}>
```

Comment:
what is `p={3}` doing?

Reply:
I didn't even notice it honestly. its from the material ui example https://material-ui.com/components/tabs/#vertical-tabs